### PR TITLE
Created static UI for purchaseOrderDetails page ( #1xhjpgd ).

### PR DESCRIPTION
**Pull request by meet-aniket**
Screenshot from 2021-12-31 10-53-40

### CODE REVIEW

File: src/views/PurchaseOrderDetail.vue [hunk 1]
```diff
@@ -0,0 +1,157 @@
+<template>
+  <ion-page>
+    <ion-header :translucent="true">
+      <ion-toolbar>
+        <ion-back-button default-href="/" slot="start"></ion-back-button>
```
Comment (azkyakhan) [inline src/views/PurchaseOrderDetail.vue:5]: use self closing tag

File: src/views/PurchaseOrderDetail.vue [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        <ion-title>{{$t("Purchase Order Details")}}</ion-title>
+        <ion-buttons slot="end">
+          <ion-button @click="addProduct"><ion-icon :icon="add"/></ion-button>
```
Comment (azkyakhan) [inline src/views/PurchaseOrderDetail.vue:8]: Improve indentation

File: src/views/PurchaseOrderDetail.vue [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+        </ion-buttons>
+      </ion-toolbar>
+    </ion-header>
+
+    <ion-content :fullscreen="true">
+      <div>
+        <ion-item lines="none">
+          <h1>{{$t("Purchase Order")}}: PO10291</h1>
+        </ion-item>
+        <ion-item>
```
Comment (azkyakhan) [inline src/views/PurchaseOrderDetail.vue:18]: Please refer to ionic framework documentation for correct usage of ionic UI components
https://ionicframework.com/docs/api/input

File: src/views/PurchaseOrderDetail.vue [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+          {{$t("Scan Items")}}
+          <ion-label>
+            <ion-input placeholder="Scan barcodes to receive"></ion-input>
```
Comment (azkyakhan) [inline src/views/PurchaseOrderDetail.vue:21]: Self closing tag

File: src/views/PurchaseOrderDetail.vue [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+          </ion-label>
+        </ion-item>
+        <ion-card>
+          <div class="product-info">
+            <ion-item lines="none">
+              <ion-thumbnail slot="start">
+                <img src="https://cdn.shopify.com/s/files/1/0069/7384/9727/products/test-track.jpg?v=1626255137" />
+              </ion-thumbnail>
+              <ion-label>
+                <h2>Chaz Kangeroo Hoodie-XS-Green</h2>
```
Comment (azkyakhan) [inline src/views/PurchaseOrderDetail.vue:31]: Keep it in `ion-label` and remove from `h2` tag.

File: src/views/PurchaseOrderDetail.vue [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+                <p>12203</p>
+              </ion-label>
+            </ion-item>
+            <ion-item class="product-count">
+              <ion-input type="number" value="0" min="0"></ion-input>
+            </ion-item>
+          </div>
+          <ion-item class="border-top">
+            <ion-button color="primary" slot="start" fill="outline">
```
Comment (azkyakhan) [inline src/views/PurchaseOrderDetail.vue:40]: Why color primary is given here ? Default color is already primary.

File: src/views/PurchaseOrderDetail.vue [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+              {{ $t("ReceiveAll") }}
+            </ion-button>
+            <ion-progress-bar value="1" color="primary"></ion-progress-bar>
```
Comment (azkyakhan) [inline src/views/PurchaseOrderDetail.vue:43]: Why color primary is given here ? Default color is already primary.

File: src/views/PurchaseOrderDetail.vue [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+  IonToolbar,
+  modalController,
+  alertController,
+} from '@ionic/vue';
+import { defineComponent } from 'vue';
+import { add, checkmarkDone } from 'ionicons/icons';
+import AddProductModal from '@/views/AddProductModal.vue'
+
+export default defineComponent({
+  name: "Shipment details",
```
Comment (azkyakhan) [inline src/views/PurchaseOrderDetail.vue:84]: Name of page is incorrect.

**Review by azkyakhan — CHANGES_REQUESTED** [review]

File: src/views/PurchaseOrderDetail.vue (hunk 1 above)
Comment (meet-aniket) [reply src/views/PurchaseOrderDetail.vue:5]: Changed Ma'am

File: src/views/PurchaseOrderDetail.vue (hunk 2 above)
Comment (meet-aniket) [reply src/views/PurchaseOrderDetail.vue:8]: Improved Ma'am.

File: src/views/PurchaseOrderDetail.vue (hunk 4 above)
Comment (meet-aniket) [reply src/views/PurchaseOrderDetail.vue:21]: Changed Ma'am.

File: src/views/PurchaseOrderDetail.vue (hunk 5 above)
Comment (meet-aniket) [reply src/views/PurchaseOrderDetail.vue:31]: Changed Ma'am.

File: src/views/PurchaseOrderDetail.vue (hunk 6 above)
Comment (meet-aniket) [reply src/views/PurchaseOrderDetail.vue:40]: Removed Ma'am.

File: src/views/PurchaseOrderDetail.vue (hunk 7 above)
Comment (meet-aniket) [reply src/views/PurchaseOrderDetail.vue:43]: Removed Ma'am.

File: src/views/PurchaseOrderDetail.vue (hunk 8 above)
Comment (meet-aniket) [reply src/views/PurchaseOrderDetail.vue:84]: Sorry Ma'am, changed name.

File: src/views/PurchaseOrderDetail.vue (hunk 3 above)
Comment (meet-aniket) [reply src/views/PurchaseOrderDetail.vue:23]: Implemented Ma'am.

File: src/views/PurchaseOrderDetail.vue [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+          <ion-label>{{$t("Scan Items")}}</ion-label>
+          <ion-label>
+            <ion-input placeholder="Scan barcodes to receive" />
+          </ion-label>
+        </ion-item>
+        <ion-card>
+          <div class="product-info">
+            <ion-item lines="none">
+              <ion-thumbnail slot="start">
+                <img src="https://cdn.shopify.com/s/files/1/0069/7384/9727/products/test-track.jpg?v=1626255137" />
```
Comment (ymaheshwari1) [inline src/views/PurchaseOrderDetail.vue:33]: Use Image component instead.

File: src/views/PurchaseOrderDetail.vue [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+          component: ReceivingHistoryModal
+        })
+      return modal.present();
+    },
+    async savePODetails() {
+      const alert = await alertController.create({
+        cssClass: "my-custom-class",
+        header: "Receive inventory",
+        message:
+          "Inventory can be received for purchase orders in multiple shipments. <br/><br/> Proceeding will recieve a new shipment for this purchase order but it will still be available for receiving later",
```
Comment (ymaheshwari1) [inline src/views/PurchaseOrderDetail.vue:122]: Internationalize the static text.

File: src/views/ReceivingHistoryModal.vue [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+        </ion-button>
+      </ion-buttons>
+      <ion-title>{{ $t("History") }}</ion-title>
+    </ion-toolbar>
+  </ion-header>
+  <ion-content class="ion-padding">
+    <ion-list>
+      <ion-item>
+        <ion-thumbnail slot="start">
+          <img src="https://demo-resources.hotwax.io/resources/uploads/images/product/m/h/mh01-black_main.jpg" />
```
Comment (ymaheshwari1) [inline src/views/ReceivingHistoryModal.vue:16]: Use Image component instead.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/views/PurchaseOrderDetail.vue [hunk 12]
```diff
@@ -0,0 +1,163 @@
...
+
+    <ion-content :fullscreen="true">
+      <div>
+        <ion-item lines="none">
+          <h1>{{$t("Purchase Order")}}: PO10291</h1>
+        </ion-item>
+        <ion-item>
+          <ion-label>{{$t("Scan Items")}}</ion-label>
+          <ion-label>
+            <ion-input placeholder="Scan barcodes to receive" />
```
Comment (azkyakhan) [inline src/views/PurchaseOrderDetail.vue:26]: Internationalize text of placeholder

**Review by azkyakhan — CHANGES_REQUESTED** [review]

File: src/views/PurchaseOrderDetail.vue (hunk 9 above)
Comment (meet-aniket) [reply src/views/PurchaseOrderDetail.vue:33]: Changed Sir.

File: src/views/ReceivingHistoryModal.vue (hunk 11 above)
Comment (meet-aniket) [reply src/views/ReceivingHistoryModal.vue:16]: Changed Sir.

File: src/views/PurchaseOrderDetail.vue (hunk 12 above)
Comment (meet-aniket) [reply src/views/PurchaseOrderDetail.vue:26]: Internationalized Sir.

File: src/views/PurchaseOrderDetail.vue (hunk 10 above)
Comment (meet-aniket) [reply src/views/PurchaseOrderDetail.vue:122]: Implemented Sir.

File: src/views/PurchaseOrderDetail.vue (hunk 3 above)
Comment (azkyakhan) [reply src/views/PurchaseOrderDetail.vue:23]: Please see it again

**Review by azkyakhan — CHANGES_REQUESTED** [review]

File: src/views/PurchaseOrderDetail.vue [hunk 13]
```diff
@@ -0,0 +1,172 @@
...
+          </ion-label>
+        </ion-item>
+        <ion-card>
+          <div class="product-info">
+            <ion-item lines="none">
+              <ion-thumbnail slot="start">
+                <Image src="https://cdn.shopify.com/s/files/1/0069/7384/9727/products/test-track.jpg?v=1626255137" />
+              </ion-thumbnail>
+              <ion-label>
+                <ion-label>Chaz Kangeroo Hoodie-XS-Green</ion-label>
```
Comment (azkyakhan) [inline src/views/PurchaseOrderDetail.vue:36]: Ion label cant be used inside ion-label. You have to write text without ion-label here.

**Review by azkyakhan — CHANGES_REQUESTED** [review]

File: src/views/PurchaseOrderDetail.vue [hunk 14]
```diff
@@ -0,0 +1,172 @@
...
+    async receivingHistory() {
+      const modal = await modalController
+        .create({
+          component: ReceivingHistoryModal
+        })
+      return modal.present();
+    },
+    async savePODetails() {
+      const alert = await alertController.create({
+        cssClass: "my-custom-class",
```
Comment (azkyakhan) [inline src/views/PurchaseOrderDetail.vue:121]: Why my-custom-class is used over here ?

**Review by azkyakhan — CHANGES_REQUESTED** [review]

File: src/views/PurchaseOrderDetail.vue (hunk 14 above)
Comment (meet-aniket) [reply src/views/PurchaseOrderDetail.vue]: Removed Ma'am

File: src/views/PurchaseOrderDetail.vue (hunk 13 above)
Comment (meet-aniket) [reply src/views/PurchaseOrderDetail.vue]: Written without ion-label

File: src/locales/en.json [hunk 15]
```diff
@@ -1,13 +1,23 @@
 {
```
Comment (azkyakhan) [inline src/locales/en.json:1]: Improve spacing

File: src/locales/en.json [hunk 16]
```diff
@@ -1,13 +1,23 @@
...
+  "History":"History",
+  "Inventory can be received for purchase orders in multiple shipments. Proceeding will recieve a new shipment for this purchase order but it will still be available for receiving later": "Inventory can be received for purchase orders in multiple shipments. { space } Proceeding will recieve a new shipment for this purchase order but it will still be available for receiving later",
   "item":"item",
   "items":"items",
   "Load more shipments":"Load more shipments",
   "Logout":"Logout",
+  "Proceed": "Proceed",
+  "Purchase Order":"Purchase Order",
+  "Purchase Order Details":"Purchase Order Details",
   "ReceiveAll": "ReceiveAll",
```
Comment (adityasharma7) [inline src/locales/en.json:15]: Is it intentional?
```suggestion
  "Receive All": "Receive All",
```

I can clearly see a space
https://www.figma.com/file/bVPRRw282CqGKMdbz7dciH/HC-Ionic-design-system?node-id=13525%3A356

@dt2patel

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/locales/en.json (hunk 16 above)
Comment (meet-aniket) [reply src/locales/en.json:15]: Yes Sir.

File: src/views/PurchaseOrderDetail.vue [hunk 17]
```diff
@@ -0,0 +1,168 @@
...
+            <ion-icon :icon="addOutline"/>
+          </ion-button>
+        </ion-buttons>
+      </ion-toolbar>
+    </ion-header>
+
+    <ion-content :fullscreen="true">
+      <div>
+        <ion-item lines="none">
+          <h1>{{$t("Purchase Order")}}: PO10291</h1>
```
Comment (azkyakhan) [inline src/views/PurchaseOrderDetail.vue:21]: Use `ion-label` instead of `h1` tag

**Review by azkyakhan — CHANGES_REQUESTED** [review]

File: src/locales/en.json (hunk 15 above)
Comment (meet-aniket) [reply src/locales/en.json:1]: Improved Ma'am.

File: src/views/PurchaseOrderDetail.vue (hunk 17 above)
Comment (meet-aniket) [reply src/views/PurchaseOrderDetail.vue:21]: Changed Ma'am

**Comment by azkyakhan** [comment]
LGTM

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/views/PurchaseOrderDetail.vue [hunk 18]
```diff
@@ -0,0 +1,168 @@
...
+              <ion-thumbnail slot="start">
+                <Image src="https://cdn.shopify.com/s/files/1/0069/7384/9727/products/test-track.jpg?v=1626255137" />
+              </ion-thumbnail>
+              <ion-label>
+                Chaz Kangeroo Hoodie-XS-Green
+                <p>12203</p>
+              </ion-label>
+            </ion-item>
+            <ion-item class="product-count">
+              <ion-input type="number" value="0" min="0"></ion-input>
```
[the remark this answers is not in the document]
Comment (meet-aniket) [reply src/views/PurchaseOrderDetail.vue:38]: Used Sir.

File: src/views/PurchaseOrderDetail.vue [hunk 19]
```diff
@@ -0,0 +1,168 @@
...
+            </ion-item>
+          </div>
+          <ion-item class="border-top">
+            <ion-button slot="start" fill="outline">
+              {{ $t("ReceiveAll") }}
+            </ion-button>
+            <ion-progress-bar value="1"></ion-progress-bar>
```
[the remark this answers is not in the document]
Comment (meet-aniket) [reply src/views/PurchaseOrderDetail.vue:45]: Changed Sir.

File: src/views/PurchaseOrderDetail.vue [hunk 20]
```diff
@@ -0,0 +1,168 @@
...
+  modalController,
+  alertController,
+} from '@ionic/vue';
+import { defineComponent } from 'vue';
+import { addOutline, saveOutline, timeOutline } from 'ionicons/icons';
+import ReceivingHistoryModal from '@/views/ReceivingHistoryModal.vue'
+import Image from "@/components/Image.vue";
+
+export default defineComponent({
+  name: "Purchase order details",
```
[the remark this answers is not in the document]
Comment (meet-aniket) [reply src/views/PurchaseOrderDetail.vue:87]: Changed Sir.

File: src/views/Shipment.vue [hunk 21]
```diff
@@ -35,9 +35,9 @@
               <ion-input type="number" min="0" v-model="item.quantityAccepted"></ion-input>
             </ion-item>
           </div>
-          <ion-item class="border-top" v-if="item.quantityOrdered > 0">
```
Comment (dt2patel) [inline src/views/Shipment.vue:38]: I think these changes are accidental @meet-aniket

File: src/views/PurchaseOrderDetail.vue [hunk 22]
```diff
@@ -0,0 +1,168 @@
...
+            <ion-icon :icon="timeOutline"/>
+          </ion-button>
+          <ion-button>
+            <ion-icon :icon="addOutline"/>
+          </ion-button>
+        </ion-buttons>
+      </ion-toolbar>
+    </ion-header>
+
+    <ion-content :fullscreen="true">
```
Comment (dt2patel) [inline src/views/PurchaseOrderDetail.vue:18]: Remove full screen

File: src/views/PurchaseOrderDetail.vue (hunk 3 above)
Comment (dt2patel) [reply src/views/PurchaseOrderDetail.vue:23]: This is not fixed yet

File: src/views/PurchaseOrderDetail.vue [hunk 23]
```diff
@@ -0,0 +1,168 @@
...
+          <div class="product-info">
+            <ion-item lines="none">
+              <ion-thumbnail slot="start">
+                <Image src="https://cdn.shopify.com/s/files/1/0069/7384/9727/products/test-track.jpg?v=1626255137" />
+              </ion-thumbnail>
+              <ion-label>
+                Chaz Kangeroo Hoodie-XS-Green
+                <p>12203</p>
+              </ion-label>
+            </ion-item>
```
Comment (dt2patel) [inline src/views/PurchaseOrderDetail.vue:36]: Missing chip in the middle

**Review by dt2patel — CHANGES_REQUESTED** [review]
Small changes and this is ready to merge. @meet-aniket

File: src/views/PurchaseOrderDetail.vue (hunk 22 above)
Comment (meet-aniket) [reply src/views/PurchaseOrderDetail.vue:18]: Removed Sir.

**Review by dt2patel — APPROVED** [review]

**Pull request merged**